core/shared/platform/zephyr/zephyr_platform.c

```diff
@@ -202,10 +202,7 @@ void
 os_dcache_flush()
 {
 #if defined(CONFIG_CPU_CORTEX_M7) && defined(CONFIG_ARM_MPU)
-    uint32 key;
-    key = irq_lock();
-    SCB_CleanDCache();
-    irq_unlock(key);
+    sys_cache_data_flush_all();
```

Comment:
Are the two functions (sys_cache_data_flush_all and sys_cache_instr_flush_range) supported in most zephyr versions, should we add version macro to control the code?

Reply:
I think you are right, for the first time this API was added in 3.3.0 [link](https://github.com/zephyrproject-rtos/zephyr/pull/50988).
Added changes